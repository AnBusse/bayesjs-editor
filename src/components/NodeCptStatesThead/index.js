import React from 'react';
import { statePropTypes } from 'models';
import PropTypes from 'prop-types';

const NodeCptStatesThead = ({ states }) => (
  <thead>
    <tr>
      {states.map(state => (
        <th key={state}>{state}</th>
      ))}
    </tr>
  </thead>
);

NodeCptStatesThead.propTypes = {
  states: PropTypes.arrayOf(statePropTypes).isRequired,
};

export default NodeCptStatesThead;
