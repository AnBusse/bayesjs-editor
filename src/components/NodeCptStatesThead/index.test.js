import React from 'react';
import { render } from '@testing-library/react';
import NodeCptStatesThead from './index';

const renderComponent = (props) => {
  const { container } = render(
    <NodeCptStatesThead {...props} />,
  );

  return container.firstChild;
};

describe('NodeCptStatesThead Component', () => {
  let component;
  const defaultProps = {
    states: [
      'State 1',
      'State 2',
      'State 3',
    ],
  };

  beforeEach(() => {
    component = renderComponent(defaultProps);
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
