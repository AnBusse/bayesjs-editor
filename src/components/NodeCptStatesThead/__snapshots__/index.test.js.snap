// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeCptStatesThead Component matches snapshot 1`] = `
<thead>
  <tr>
    <th>
      State 1
    </th>
    <th>
      State 2
    </th>
    <th>
      State 3
    </th>
  </tr>
</thead>
`;
